src/vs/workbench/api/common/extHostTypes.ts

```diff
@@ -1707,10 +1707,31 @@ export enum SourceControlInputBoxValidationType {
 	Information = 2
 }
 
+export class TerminalLink implements vscode.TerminalLink {
+	constructor(
+		public startIndex: number,
+		public length: number,
+		public tooltip?: string
+	) {
+		if (typeof startIndex !== 'number') {
```

Comment:
maybe additional (more useful) checks would be for index and length to be positive